src/thor/costmatrix.cc

```diff
@@ -318,6 +321,8 @@ void CostMatrix::Initialize(
 
   locs_count_[MATRIX_FORW] = source_locations.size();
   locs_count_[MATRIX_REV] = target_locations.size();
+  astar_heuristics_[MATRIX_FORW].resize(target_locations.size());
```

Comment:
for each **source** we create one heuristic for the **reverse tree** and vice versa